File: compilers/lab1/token.go
package main

import "fmt"

type Token struct {
	Type     TokenType
	Value    string
	Position int
}

func NewToken(tokenType TokenType, value string, position int) Token {
	return Token{
		Type:     tokenType,
		Value:    value,
		Position: position,
	}
}

func (t Token) String() string {
	return fmt.Sprintf("Token(Type: %s, Value: '%s', Position: %d)", t.Type, t.Value, t.Position)
}

File: compilers/lab1/Lexer/Lexer/Token.cs
namespace Lexer
{
    public class Token
    {
        public TokenType Type { get; }
        public string Value { get; }
        public int Position { get; }
        public Token(TokenType type, string value, int position)
        {
            Type = type;
            Value = value;
            Position = position;
        }
        public override string ToString()
        {
            return $"Token(Type: {Type}, Value: '{Value}', Position: {Position})";
        }
    }
}